v2/internal/test/fake/bundleendpoint.go

```diff
@@ -0,0 +1,138 @@
+package fake
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"net/http"
+	"sync"
+	"testing"
+
+	"github.com/spiffe/go-spiffe/v2/bundle/spiffebundle"
+	"github.com/spiffe/go-spiffe/v2/internal/test"
+	"github.com/spiffe/go-spiffe/v2/internal/x509util"
+	"github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig"
+	"github.com/spiffe/go-spiffe/v2/svid/x509svid"
+	"github.com/stretchr/testify/assert"
+)
+
+type BundleEndpoint struct {
+	tb     testing.TB
+	wg     sync.WaitGroup
+	addr   net.Addr
+	server *http.Server
+	// Root certificates used by clients to verify server certificates.
+	rootCAs *x509.CertPool
+	// TLS configuration used by the server.
+	tlscfg *tls.Config
+	// SPIFFE bundle returned by this BundleEndpoint.
+	bundle *spiffebundle.Bundle
+}
+
+type BundleEndpointOption interface {
+	apply(*BundleEndpoint)
+}
+
+// BEOption groups the available options for a BundleEndpoint.
+var BEOption = new(bundleEndpointOption)
```

Comment:
Hmm. This feels a little weird to me. If we're worried about namespacing the options, we should move this all into it's own package instead. Maybe `test/fakebundleendpoint`?

Reply:
I knew this was going to be controversial :stuck_out_tongue: 
This was an attempt  to avoid creating one package for having just one file...for some reason that kind of packages feels weird to me, maybe because I still don't have enough experience using Go. Anyway, I don't have strong feelings about this, so I will update this.